contracts/token/ERC1155/ERC1155.sol

```diff
@@ -0,0 +1,234 @@
+pragma solidity ^0.5.0;
+
+import "./IERC1155.sol";
+import "./IERC1155TokenReceiver.sol";
+import "../../math/SafeMath.sol";
+import "../../utils/Address.sol";
+import "../../introspection/ERC165.sol";
+
+/**
+ * @title Standard ERC1155 token
+ *
+ * @dev Implementation of the basic standard multi-token.
+ * See https://eips.ethereum.org/EIPS/eip-1155
+ * Originally based on code by Enjin: https://github.com/enjin/erc-1155
+ */
+contract ERC1155 is ERC165, IERC1155
```

Comment:
The system relies on the events to know which `id`s are available. Nevertheless, it would be nice to have in the blockchain a variable that reflects which `id`s has been minted. Additionally, shouldn't it revert at the start of a transfer (e.g.) if the `id` hasn't been minted?

Reply:
I don't think so, since this will limit the use of counterfactual IDs, or IDs which are presumed by the contract to have some meaning, but do not actually hold any supply yet. For example, I work at Gnosis, and we have a contract which makes extensive use of IDs which may not exist on-chain yet. In general, if the IDs are less than completely arbitrary (i.e. they encode some information), then the use of them in a counterfactual manner would be limited.

In any case, it would revert _anyway_ at a start of a transfer if the `id` hasn't been minted, as there wouldn't be addresses holding any balance with it.